### DISCUSSION TITLE
Difference between metrics for create_model(cross_validation=False) and confusion matrix

### DISCUSSION THREAD

**Discussion by thewiredbear** [pycaret/pycaret#4002] [Q&A]
Okay, so to the best of my knowledge, if we choose the option cross_validation=False for create_model or compare_models then the metrics are calculated for holdout/test set. Plot_model also by default runs on the holdout set. Now essentially if I am using cross_validation=False, then the metrics from create_model/compare_models should be same as plot_model.

So I am testing pycaret with the Kaggle Telco churn dataset, and I see a huge difference between (let's say for example, Recall) create_model output and metrics manually calculated from confusion matrix.

<img width="731" alt="image" src="https://github.com/pycaret/pycaret/assets/16281860/7195a167-edcd-4626-8b9e-9cb566c7af5a">

As you can see in the image above, create_model gives recall of 74% but confusion matrix depicts a recall of around 55%. Why such a big difference?

**Comment by celestinoxp** [comment]
@thewiredbear , I'm a little confused and I'm not sure what's going on... shall we investigate? So maybe the first step is to check which versions you are using... pycaret version? scikit-learn version?
If I change the scikit-learn version to a more recent one, are the values ​​correct? no? Can you open an issue or if you can collaborate, can you create a pull-request?
